src/generic/splitter.cpp

```diff
@@ -175,11 +179,23 @@ void wxSplitterWindow::OnPaint(wxPaintEvent& WXUNUSED(event))
 {
     wxPaintDC dc(this);
 #ifdef __WXOSX__
-    // as subpanels might have a transparent background we must erase the background
-    // at least on OSX, otherwise traces of the sash will remain
-    // test with: splitter sample->replace right window
-    dc.Clear();
-#endif
+  #if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_16
```

Comment:
Do we even need this preprocessor check? It does no harm, of course, but it looks like the run-time check would be enough and would simplify the structure a little bit.

Reply:
I think it is best to do this preprocessor check, because then we don't interfere with the OS' built-in backwards compatibility layer. An app built against SDK 10.14 will be treated by the OS as such, so we shouldn't interfere IMHO

Reply:
Sorry, I missed this question. I concur that the check should be there: apps built against pre-10.14 SDK may (and almost certainly do!) behave differently w.r.t. backing layers' change of behavior in 10.14.